File: Java_ChildrenCompetitionSockets/CompetitionPersistence/src/main/java/competition/persistence/IUserRepository.java
package competition.persistence;


//public interface IUserRepository <ID, T extends HasId<ID>> extends IRepository<ID, T>{
//    void setCurrentUser(T entity);
//    T getCurrentUser();
//    T findByUsername(String username);
//}

import competition.model.User;

public interface IUserRepository extends IRepository<Integer, User> {
    void setCurrentUser(User entity);
    User getCurrentUser();
    User findByUsername(String username);
}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/persistence/IUserRepository.cs
using CSharp_ChildrenCompetitionGUI.model;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public interface IUserRepository : IRepository<int, User>
    {
        void setCurrentUser(User entity);
        User getCurrentUser();
        User findByUsername(string username);
    }
}